nomad/structs/acl.go

```diff
@@ -363,7 +363,7 @@ func (a *ACLToken) Validate(minTTL, maxTTL time.Duration, existing *ACLToken) er
 		if existing.ExpirationTTL != a.ExpirationTTL {
 			mErr.Errors = append(mErr.Errors, errors.New("cannot update expiration TTL"))
 		}
-		if existing.ExpirationTime != a.ExpirationTime {
+		if !existing.ExpirationTime.Equal(*a.ExpirationTime) {
```

Comment:
Context: I found that every time I tried to update a token via http api (to change its policyIDs, for example), I would receive a 500 error saying the expirationTime couldn't be updated / must match.

Traced it back to the fact that you oughtn't do `==` comparisons of time in Go (and many other languages)  because of monotonic clock time: https://pkg.go.dev/time#hdr-Monotonic_Clocks

Go luckily has a time.Equal() built-in method.